contracts/core/AuxiliaryBlockStore.sol

```diff
@@ -378,4 +441,47 @@ contract AuxiliaryBlockStore is BlockStore {
 
         isAncestor_ = currentCheckpoint.blockHash == _ancestor;
     }
+
+    /**
+     * @notice Stores the kernel observation for the dynasty number.
+     *         It also activates already reported kernel
+     *
+     * @param _blockHash The checkpoint that shall be finalized.
+     */
+    function updateKernel(bytes32 _blockHash)
+        private
+    {
+
+        bytes32 openKernelHash =
+            kernelGateway.getOpenKernelHash(currentDynasty);
+
+        // Check if any kernel is reported for activation.
+        if(openKernelHash != bytes32(0)) {
+
+            address[] memory updatedValidators;
+            uint256[] memory updatedWeights;
+
+            (updatedValidators, updatedWeights) =
+                kernelGateway.getUpdatedValidators(openKernelHash);
+
+            /*
+             * Report polling place about new validators and new finalized
+             * block heights.
+             */
+            PollingPlaceInterface(pollingPlace).updateMetaBlockHeight(
+                updatedValidators,
+                updatedWeights,
+                originBlockStore.getCurrentDynasty(),
+                currentDynasty
+            );
+
+            // Update the active kernel hash.
+            kernelGateway.activateKernel(openKernelHash);
+        }
+
+        // Store the kernel observations for the block hash.
+        kernelHashes[_blockHash] = kernelGateway.getActiveKernelHash();
+
+    }
```

Comment:
I don't quire understand `updateKernel`. I also don't quite understand why it is called where it is called and why it's called with the target as its argument 🤔
Can you help me by adding more explanation to the code? Also regarding `openKernelHash`.

Reply:
Added the comments.